functions/src/scripts/denormalize.ts

```diff
@@ -0,0 +1,48 @@
+// Helper functions for maintaining the relationship between fields in one set of documents and denormalized copies in
```

Comment:
I like having helpers that abstract out the work to be done, but on the other hand, it's often simpler to follow if there're fewer layers of abstraction.

I'm not sure this is better than having one function for each of contracts, comments, and answers that updates the applicable documents for a single user (and then called for every user).